scipy/stats/_stats_py.py

```diff
@@ -672,14 +679,18 @@ def tvar(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
 
     """
     a = asarray(a)
-    a = a.astype(float)
+    if not np.issubdtype(a.dtype, np.inexact):
+        a = a.astype(np.float64)
     if limits is None:
         return a.var(ddof=ddof, axis=axis)
     am = _mask_to_limits(a, limits, inclusive)
-    amnan = am.filled(fill_value=np.nan)
+    amnan = am.filled(fill_value=_get_nan(a))
     return np.nanvar(amnan, ddof=ddof, axis=axis)
```

Comment:
Changed `tmean` to use the same approach as `tvar` and `tstd`.

Reply:
Ok, but do you know why those use `fill` and `nanmean` rather than just using the masked array function? `min` and `max` do that, so they are still completely different : )

Again, could be a separate PR, since it wasn't in scope of this PR to clean these up. It's just surprising to me that SciPy would be so creative here.

Reply:
When a slice is fully masked, calling `np.ma.mean` or `np.ma.var` puts a `np.ma.masked` singleton in that slice. The data in it is zero so if we do `return res.data` on it, it will return `0` instead of `np.nan`. We will need a special case to handle that. Given that, filling in nans looks like a cleaner approach. What do you think?

Reply:
Yeah, I know about those masked constants. That's why I was thinking it might need some more dedicated experimentation in a separate PR.

> Given that, filling in nans looks like a cleaner approach. What do you think?

Yes, that is what I was thinking for the same reason. I wondered why not just fill in NaNs from the outset rather than the intermediate masked arrays. That's what I'd prefer.

Reply:
I guess intermediate masked arrays help `tmax` and `tmin` where `ma.min` and `ma.max` are called directly.

Reply:
Yeah but that's just because they still use a different approach - _why_ should they even use `ma.min`/`ma.max` instead of `nanmin`/`nanmax`?

Reply:
Ah, makes sense. Changed it to directly put NaNs instead of converting to masked arrays.